### PR TITLE
Implement Add Photos to Artwork Functionality

### LINKED ISSUE #123: Add Photos to Previously Discovered Artworks
On the `Artwork` page, allow the user to take additional photos and upload them to that artwork by clicking on the camera button at the bottom of the page.

## User Flow
1. Go to artwork page.
2. Click on camera button.
3. Take photos.
- View Photos in photo library before submitting.
4. Press submit button.
5. Images uploaded.
6. Get rerouted to the artwork page.

### PR DESCRIPTION
**Pull request by gwholland3**
## Summary
You should now be able to use the camera button on an artwork page to take/upload new pictures and submit them for that artwork.

I used the same `Camera` component that we use for our main `/camera` page, except modified it so that it takes an `onSubmit` function, allowing it to be used slightly differently in both cases.

## Testing
Start up the frontend and the backend, log in, and click on any artwork in your portfolio. Click on the camera button, which should take you to the camera interface. Then take/upload any number of photos, and click the next arrow.

If everything goes well, you should be taken right back to the artwork page. To check that your photos actually went through, use the GraphQL interface to query for that artwork's pictures, and make sure there are more than one.

Fixes #123

### CODE REVIEW

File: frontend/src/pages/Artwork/Artwork.tsx [hunk 1]
```diff
@@ -26,17 +28,38 @@ import MetricBadge from "../../components/MetricBadge/MetricBadge";
...
+        id: $artworkId
+        pictures: $pictures_to_add
+      }
+    ) {
+      artwork {
+        id
+      }
+    }
+  }
+`;
```
Comment (Waidhoferj) [inline frontend/src/pages/Artwork/Artwork.tsx]: Can this be moved to the `gql.ts` file where all of the other mutations and queries are defined? This keeps the component file a little more concise and divides concerns.

File: frontend/src/pages/Artwork/Artwork.tsx [hunk 2]
```diff
@@ -69,6 +92,17 @@ export default function Artwork() {
...
+    const payload = {
+      artworkId: id, 
+      pictures_to_add: images,
+    };
+    addPhotos({ variables: payload }).then((res) => {
+      clearLibrary();
+      push("/artwork/" + res.data.updateArtwork.artwork.id);
+    });
+  }
+
```
Comment (Waidhoferj) [inline frontend/src/pages/Artwork/Artwork.tsx:88]: Great use of the usePhotoLibrary hook!

File: frontend/src/pages/Camera/Camera.jsx [hunk 3]
```diff
@@ -25,7 +26,7 @@ export default function Camera() {
             exit={{ opacity: 0, transform: "translateY(100vh)" }}
             transition={{ ease: ["easeInOut"], duration: 0.4 }}
           >
-            <PhotoGallery onHide={() => history.push("/camera")} />
+            <PhotoGallery onHide={() => history.goBack()} />
```
Comment (Waidhoferj) [inline frontend/src/pages/Camera/Camera.jsx:29]: Ah yeah this makes it way more extensible, nice edit!

File: frontend/src/pages/Camera/components/CameraCapture.jsx [hunk 4]
```diff
@@ -13,10 +12,9 @@ const videoConstraints = {
   facingMode: "environment",
 };
 
-export default function CameraCapture({ onShowPhotos, onImageCapture }) {
+export default function CameraCapture({ onShowPhotos, onImageCapture, onSubmit}) {
```
Comment (Waidhoferj) [inline frontend/src/pages/Camera/components/CameraCapture.jsx:15]: Really appreciate how extensible the Camera component is now! Great adaptation of the component

**Review by Waidhoferj — APPROVED** [review]
Looks good! I'd move the mutation to another file, but thats up to you. The code and the tests run as expected.

File: frontend/src/pages/Artwork/Artwork.tsx (hunk 1 above)
Comment (gwholland3) [reply frontend/src/pages/Artwork/Artwork.tsx]: Sounds good, will do!

File: frontend/src/pages/Camera/Camera.jsx [hunk 5]
```diff
@@ -7,14 +7,15 @@ import PhotoGallery from "./components/PhotoGallery";
 import { Route, useHistory } from "react-router-dom";
 import usePhotoLibrary from "../../hooks/usePhotoLibrary";
 
-export default function Camera() {
+export default function Camera({onSubmit}) {
```
Comment (Braden50) [inline frontend/src/pages/Camera/Camera.jsx:10]: Interesting edit, I'll have to get you or John to explain its effects

File: backend/mutations.py [hunk 6]
```diff
@@ -526,8 +526,8 @@ def mutate(self, info, artwork_data=None):
             artwork.artist = artwork_data.artist
         if artwork_data.description:
             artwork.description = artwork_data.description
-        if artwork_data.picture_to_add:
-            artwork.pictures.append(artwork_data.picture_to_add)
+        if artwork_data.pictures:
+            artwork.pictures.extend(artwork_data.pictures)
```
Comment (Braden50) [inline backend/mutations.py:530]: Good design decision!

File: frontend/src/pages/Artwork/gql.ts [hunk 7]
```diff
@@ -1,5 +1,23 @@
 import { gql } from "@apollo/client";
 
+export const ADD_PHOTOS = gql`
```
Comment (Braden50) [inline frontend/src/pages/Artwork/gql.ts:3]: Good to see this mutation being used and working on the front end!

**Review by Braden50 — APPROVED** [review]
Everything seems to look good to me. Testing with the API was difficult though with the long strings, so I hope I was able to do it correctly. 

We should probably add the ability to upload pictures that don't come directly from the camera since this feature will most likely be used when users are away from the art. This can be a separate issue if necessary.

**Comment by gwholland3** [comment]
> We should probably add the ability to upload pictures that don't come directly from the camera since this feature will most likely be used when users are away from the art. This can be a separate issue if necessary.

@Braden50 I believe the user is already able to do that, if they use the upload photos button in the Camera component?

**Comment by Braden50** [comment]
> > We should probably add the ability to upload pictures that don't come directly from the camera since this feature will most likely be used when users are away from the art. This can be a separate issue if necessary.
> 
> @Braden50 I believe the user is already able to do that, if they use the upload photos button in the Camera component?

You're right, I'm not sure why I didn't see it. Maybe I just didn't pay attention close enough to see the button. Good work!

File: frontend/src/pages/Artwork/Artwork.tsx [hunk 8]
```diff
@@ -117,8 +134,8 @@ export default function Artwork() {
               <button onClick={() => push("/artwork/" + id + "/report")}>
                 <AlertCircle />
               </button>
-              <button>
-                <Camera />
+              <button onClick={() => push("/artwork/" + id + "/add-photos")}>
+                <CamIcon />
```
Comment (kyleshare) [inline frontend/src/pages/Artwork/Artwork.tsx:138]: Good call with this redirect

File: frontend/src/pages/Artwork/Artwork.tsx [hunk 9]
```diff
@@ -147,6 +164,9 @@ export default function Artwork() {
           postResolver={artPostResolver}
         />
       </Route>
+      <Route exact path="/artwork/:id/add-photos">
+        <Camera onSubmit={uploadImages}/>
+      </Route>
```
Comment (kyleshare) [inline frontend/src/pages/Artwork/Artwork.tsx:169]: Good use of uploadImages

**Pull request merged**